plugins/policy-recommendation/policy_recommendation_job.py

```diff
@@ -0,0 +1,856 @@
+            kind = "ClusterNetworkPolicy",
+            api_version = "crd.antrea.io/v1alpha1",
+            metadata = kubernetes.client.V1ObjectMeta(
+                name = np_name,
+            ),
+            spec = antrea_crd.NetworkPolicySpec(
+                tier = "Platform",
+                priority = DEFAULT_POLICY_PRIORITY,
+                applied_to = [antrea_crd.NetworkPolicyPeer(
+                    namespace_selector = kubernetes.client.V1LabelSelector(
+                        match_labels = {
+                            "kubernetes.io/metadata.name":ns
+                        }
+                    )
+                )],
+                egress = [antrea_crd.Rule(
+                    action = "Allow",
+                    to = [antrea_crd.NetworkPolicyPeer(
+                        pod_selector = kubernetes.client.V1LabelSelector())]
+                )],
+                ingress = [antrea_crd.Rule(
+                    action = "Allow",
+                    _from = [antrea_crd.NetworkPolicyPeer(
+                        pod_selector = kubernetes.client.V1LabelSelector())]
+                )],
+            )
+        )
+        policies.append(dict_to_yaml(acnp.to_dict()))
+    return policies
+
+def generate_sql_query(table_name, limit, start_time, end_time, unprotected):
+    sql_query = "SELECT {} FROM {}".format(", ".join(FLOW_TABLE_COLUMNS), table_name)
+    if unprotected:
+        sql_query += " WHERE ingressNetworkPolicyName == '' AND egressNetworkPolicyName == ''"
+    else:
+        # Select user trusted denied flows when unprotected equals False
+        sql_query += " WHERE trusted == 1"
+    if start_time:
+        sql_query += " AND flowStartSeconds >= '{}'".format(start_time)
+    if end_time:
+        sql_query += " AND flowEndSeconds < '{}'".format(end_time)
+    sql_query += " GROUP BY {}".format(", ".join(FLOW_TABLE_COLUMNS))
+    if limit:
+        sql_query += " LIMIT {}".format(limit)
+    return sql_query
+
+def read_flow_df(spark, db_jdbc_address, sql_query, rm_labels):
+    flow_df = spark.read \
+        .format("jdbc") \
+        .option("driver", "ru.yandex.clickhouse.ClickHouseDriver") \
+        .option("url", db_jdbc_address) \
+        .option("user", os.getenv("CH_USERNAME")) \
+        .option("password", os.getenv("CH_PASSWORD")) \
+        .option("query",  sql_query)\
+        .load()
+    if rm_labels:
+        flow_df = flow_df.withColumn('sourcePodLabels', udf(remove_meaningless_labels, StringType())("sourcePodLabels"))\
+            .withColumn('destinationPodLabels', udf(remove_meaningless_labels, StringType())("destinationPodLabels"))\
+            .dropDuplicates(['sourcePodLabels', 'destinationPodLabels'])
+    flow_df = flow_df.withColumn('flowType', udf(get_flow_type, StringType())("destinationServicePortName", "destinationPodLabels"))
+    return flow_df
+
+def write_recommendation_result(spark, result, recommendation_type, db_jdbc_address, table_name, id):
+    if not id:
```

Comment:
nit (perhaps to be addressed in a future PR): id is a reserved keyword in python. In this case it will be correctly interpreted, but using it is a risk for a maintainability perspective (e.g.: we change the param name to something else, no error is thrown when running the code, but "if not id" will always be false!)

Reply:
Thank Salvatore, that's a fair concern. I rename this parameter to `recommendation_id_input` instead.